source/grid/tria.cc

```diff
@@ -2068,11 +2068,22 @@ namespace internal
             // need to allow that
             if (boundary_line->boundary_id != numbers::internal_face_boundary_id)
               {
-                AssertThrow (! (line->boundary_id() == numbers::internal_face_boundary_id),
-                             ExcInteriorLineCantBeBoundary(line->vertex_index(0),
-                                                           line->vertex_index(1),
-                                                           boundary_line->boundary_id));
-                line->set_boundary_id (boundary_line->boundary_id);
+                if (line->boundary_id() == numbers::internal_face_boundary_id)
+                  {
+                    if (boundary_line->manifold_id == numbers::flat_manifold_id)
+                      {
+                        AssertThrow (! (line->boundary_id() == numbers::internal_face_boundary_id),
```

Comment:
I mixed up my conditions, but what I meant is that the code now looks like this:
```
  if (condition)
    {    // here we know that  condition==true
      if (some_other_condition)
        {
           AssertThrow (!condition, ...);    // same condition as above, so will ALWAYS fail
```
Or do I miss something?